pyctcdecode/language_model.py

```diff
@@ -271,6 +287,84 @@ def score(
         lm_score = self.alpha * lm_score * LOG_BASE_CHANGE_FACTOR + self.beta
         return lm_score, end_state
 
+    @property
+    def serializable_attrs(self) -> Dict[str, Any]:
+        """Get a dictionary of the attributes to serialize to json."""
+        json_attrs = {}
+        for attr in LanguageModel.json_attrs:
+            val = getattr(self, attr)
+            if val is None:
+                raise ValueError(f"attribute {attr} not found. Cannot serialize")
+            json_attrs[attr] = val
+        return json_attrs
+
+    def save_to_dir(self, filepath: str) -> None:
+        """Save to a directory."""
+        json_attrs = self.serializable_attrs
+        json_attr_path = os.path.join(filepath, "attrs.json")
+        unigrams_path = os.path.join(filepath, "unigrams.json" "")
+        kenlm_filename = os.path.split(self._kenlm_model.path.decode("utf-8"))[1]
+        kenlm_path = os.path.join(filepath, kenlm_filename)
+
+        with open(json_attr_path, "w") as fi:
+            json.dump(json_attrs, fi)
+
+        with open(unigrams_path, "w") as fi:
+            json.dump(sorted(self._unigram_set), fi)
+
+        logger.info(
+            "copying kenlm model from %s to %s. " "This may take some time",
+            self._kenlm_model.path,
+            kenlm_path,
+        )
+        shutil.copy2(self._kenlm_model.path, kenlm_path)
+
+    @staticmethod
+    def parse_directory_contents(filepath: str) -> Dict[str, str]:
+        """Check the contents of a directory for the correct files."""
+        contents = os.listdir(filepath)
+        if len(contents) != 3:
+            raise ValueError(
```

Comment:
hah, i already see problems incoming where people open the dir in finder on mac and then this yacks cause of hidden files